app/code/Magento/Inventory/Controller/Adminhtml/Source/Save.php

```diff
@@ -134,8 +115,8 @@ private function processSave($sourceId, array $requestData)
             /** @var SourceInterface $source */
             $source = $this->sourceFactory->create();
         }
-        $this->dataObjectHelper->populateWithArray($source, $requestData, SourceInterface::class);
-        $source = $this->carrierRequestDataHydrator->hydrate($source, $requestData);
+        $this->dataObjectHelper->populateWithArray($source, $requestData['general'], SourceInterface::class);
+        $source = $this->carrierRequestDataHydrator->hydrate($source, $requestData['general']);
```

Comment:
Need to discuss how to do it better
This code related only to controller (for specific form)